specs/protocol/revenue-sharing.md

```diff
@@ -0,0 +1,77 @@
+# Revenue Sharing
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+  - [Definitions](#definitions)
+  - [`RevenueSharer` predeploy](#revenuesharer-predeploy)
+    - [Invariants](#invariants)
+    - [Deploying `RevenueSharer`](#deploying-revenuesharer)
+      - [Existing Chains](#existing-chains)
+    - [Execution](#execution)
+  - [Simplified L1 Data Fee Expenditure](#simplified-l1-data-fee-expenditure)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+# Overview
+
+Revenue sharing is the process by which chains in the superchain contribute a portion of their revenue or profit to a beneficiary such as the Optimism collective. They may do this in return for support with the OP Stack and/or other benefits. 
+
+
+## Definitions
+| Term   | Name         | Definition  |
+| -------|--------------| ----------- |
+| $d$ | L1 Data Fee Revenue    | ETH transferred to the sequencer with L2 transactions to cover estimated L1 Data Fees(see below). Accumulates to `L1FeeVault`.
+| $e$ | L1 Data Fee Expenditure| ETH spent by the batcher on L1 to make transaction data available
+| $b$ | L2 Base Gas Revenue         | Base fee portion of L2 Gas Fee (ETH transferred to the sequencer to cover execution of L2 transactions). Accumulates to `BaseFeeVault`.
+| $q$ | L2 Priority Gas Revenue         | Priority fee portion of L2 Gas Fee (ETH transferred to the sequencer to cover execution of L2 transactions). Accumulates to `SequencerFeeVault`.
+| $g$ | L2 Gas Revenue         | $b + q$
+| $r$ | Sequencer Revenue      | $d + b + q$
+| $p$ | Sequencer Profit       | $r - e$
+| $s$ | Revenue share due to Optimism Collective | $\max(0.15r,0.025p)$
+
+## `RevenueSharer` predeploy
+Revenue sharing is achieved through an L2 [predeploy](./predeploys.md) contract `RevenueSharer` with address `0x4200000000000000000000000000000000000024`.
+
+### Invariants
```

Comment:
The vaults must be configured to withdraw to this predeploy?

Reply:
This is relevant to [your other comment.](https://github.com/ethereum-optimism/specs/pull/115#discussion_r1566616859)

The model I initially had in mind was to hardcode the `RevenueShare(r)` as the output of the `FeeVaults`. To my mind, this represents a _greater_ level of enshrinement of revenue sharing, which I believe is our overarching goal. But it does have some downsides:
1. It could be offputting or confusing to any chain operator who isn't sure about doing revenue sharing in the first place. As you say, from an open source perspective it's nice to include this set of users.
2. It reduces overall configurability over how fees can flow around. Although we can ensure that the `RevenueShare(r)` can still be configured to send _all_ funds to e.g. the chain operator or any other wallet, this approach does tie all three FeeVaults together (not possible to withdraw from one without withdrawing from the others).

We can make a different choice and make the the output of the FeeVaults freely configurable. This represents a somewhat _lesser_ enshrinement of revenue sharing. The downside could be that:
1. We risk not achieving our overall goal. Chains can easily bypass the `RevenueShare(r)` predeploy, leaving it dangling there doing nothing,
2. There is more opportunity to deploy a "bad" configuration which could lead to incorrect behaviour. Of course this may be addressable through documentation and/or runtime checks.

Reply:
From slack thread: would be nice to have a simple way to opt out of revenue sharing via a flag in the deploy configuration. When opting out, the existing/legacy behaviour can prevail.